strongbox-cron/strongbox-cron-api/src/main/java/org/carlspring/strongbox/cron/services/impl/CronTaskConfigurationServiceImpl.java

```diff
@@ -1,92 +1,93 @@
     {
-        beanFactory = context.getAutowireCapableBeanFactory();
+        CronTasksConfigurationDto cronTasksConfiguration = getTasksConfigurationDto();
+
+        for (CronTaskConfigurationDto dto : cronTasksConfiguration.getCronTaskConfigurations())
+        {
+            if (dto.isOneTimeExecution())
+            {
+                continue;
+            }
+            
+            cronJobSchedulerService.scheduleJob(dto);
+            
+            if (dto.shouldExecuteImmediately())
+            {
+                cronJobSchedulerService.executeJob(dto);
+            }
+        }
     }
 
     public void saveConfiguration(CronTaskConfigurationDto configuration)
-            throws Exception
+        throws Exception
     {
         logger.debug("CronTaskConfigurationService.saveConfiguration()");
 
-        if (!configuration.contains("cronExpression"))
-        {
-            throw new CronTaskException("cronExpression property does not exists");
-        }
+        Class c = Class.forName(configuration.getProperty("jobClass"));
+        Object classInstance = c.newInstance();
 
-        cronTaskDataService.save(configuration);
+        logger.debug("> " + c.getSuperclass().getCanonicalName());
 
-        if (configuration.contains("jobClass"))
+        if (!(classInstance instanceof AbstractCronJob))
         {
-            Class c = Class.forName(configuration.getProperty("jobClass"));
-            Object classInstance = c.newInstance();
-
-            logger.debug("> " + c.getSuperclass().getCanonicalName());
-
-            if (!(classInstance instanceof AbstractCronJob))
-            {
-                throw new CronTaskException(c + " does not extend " + AbstractCronJob.class);
-            }
-
-            beanFactory.autowireBean(classInstance);
-            ((AbstractCronJob) classInstance).beforeScheduleCallback(configuration);
+            throw new CronTaskException(c + " does not extend " + AbstractCronJob.class);
+        }
 
-            cronJobSchedulerService.scheduleJob(configuration);
+        if (!configuration.isOneTimeExecution())
+        {
+            cronTaskDataService.save(configuration);
+        }   
 
-            if (configuration.shouldExecuteImmediately())
-            {
-                cronJobSchedulerService.executeJob(configuration);
-            }
+        cronJobSchedulerService.scheduleJob(configuration);
+        if (configuration.isOneTimeExecution() || configuration.shouldExecuteImmediately())
+        {
+            cronJobSchedulerService.executeJob(configuration);
         }
 
         cronTaskEventListenerRegistry.dispatchCronTaskCreatedEvent(configuration.getName());
     }
 
     public void deleteConfiguration(String cronTaskConfigurationName)
-            throws SchedulerException,
-                   CronTaskNotFoundException,
-                   ClassNotFoundException
+        throws SchedulerException,
+        CronTaskNotFoundException,
+        ClassNotFoundException
```

Comment:
Cosmetic: Would you mind restoring the original indentation here? Thanks! :)

Reply:
done